roles/service-dehydrated/tasks/main.yml

```diff
@@ -0,0 +1,127 @@
+  apt: name={{ dehydrated_dependencies }}
+
+- name: Checkout dehydrated from github
+  git:
+    repo: "{{ dehydrated_repo_url }}"
+    update: "{{ dehydrated_update }}"
+    dest: "{{ dehydrated_install_root }}"
+    version: "{{ dehydrated_version }}"
+
+- name: Checkout pdns_api.sh from github
+  git:
+    repo: "{{ pdns_api_repo_url }}"
+    update: "{{ pdns_api_update }}"
+    dest: "{{ dehydrated_install_root }}/pdns_api"
+    version: "{{ pdns_api_version }}"
+
+- name: Create /etc/dehydrated
+  file: dest=/etc/dehydrated state=directory owner=root group=root mode=0700
+
+- name: Generate dehydrated config
+  template:
+    dest: /etc/dehydrated/config
+    src: config.j2
+    owner: root
+    group: root
+    mode: 0600
+
+- name: Generate dehydrated domains.txt
+  copy:
+    dest: /etc/dehydrated/domains.txt
+    content: "{{ dehydrated_domains }}"
+    owner: root
+    group: root
+    mode: 0600
+  notify: run dehydrated
+
+- import_tasks: domain_config.yml
+
+- name: Generate deploycert.sh
+  template:
+    src: deploycert.j2
+    dest: /etc/dehydrated/deploycert.sh
+    owner: root
+    group: root
+    mode: "0700"
+  when: dehydrated_deploycert is defined
+
+- name: Remove deploycert.sh
+  file: dest=/etc/dehydrated/deploycert.sh state=absent
+  when: dehydrated_deploycert is not defined
+
+- name: Install cronjob
+  cron:
+    name: dehydrated-renew
+    minute: "{{ 59|random(seed=inventory_hostname) }}"
+    hour: "{{ 4|random(seed=inventory_hostname) }}"
+    user: root
+    job: "{{ dehydrated_install_root }}/dehydrated -c"
+    cron_file: dehydrated
+    state: "{{ 'present' if dehydrated_cronjob else 'absent' }}"
+
+- import_tasks: systemd.yml
+
+# /opt/dehydrated/dehydrated --register --accept-terms
+- name: Check if already registered
+  stat:
+    path: "/etc/dehydrated/accounts/{{ ((dehydrated_ca + '\n')|b64encode).rstrip('=').replace('+', '-').replace('/', '_') }}"
+  register: ca_stat
+
+- block:
+    - name: "assert dehydrated_accept_letsencrypt_terms is true"
+      assert:
+        that: dehydrated_accept_letsencrypt_terms
+
+    - name: Register to CA
+      command: "{{ dehydrated_install_root }}/dehydrated --register --accept-terms"
+  # \end block register
+  when: "not ca_stat.stat.exists or (ca_stat.stat.isdir is defined and not ca_stat.stat.isdir)"
+
+- meta: flush_handlers
+
+
+- name: Add the cert user for distributing certs
+  user:
+    name: cert
+
+- name: Create cert/bin directory if it does not exist
+  file:
+    path: /home/cert/bin
+    state: directory
+    owner: cert
+    group: cert
+    mode: '0700'
+
+- name: Create certificates directory if it does not exist
+  file:
+    path: /home/cert/certificates
+    state: directory
+    owner: cert
+    group: cert
+    mode: '0700'
+
+- name: Create .ssh directory if it does not exist
+  file:
+    path: /home/cert/.ssh
+    state: directory
+    owner: cert
+    group: cert
+    mode: '0700'
+
+- name: copy authorized_keys
+  copy:
```

Comment:
Die authorized_keys würde ich mit einem Template erzeugen dann muss man nicht für jeden Server die ganzen Einschränkungen von Hand einfügen und hat bei einer Änderung an den Parametern weniger Arbeit.

Reply:
Ich sehe gerade es gibt auch ein Modul um die Keys zu managen:
https://docs.ansible.com/ansible/latest/modules/authorized_key_module.html

Reply:
Ja, habe ich schon gesehen. Aber wo soll denn der Key herkommen? Public key im pass?

Reply:
Ok, habe das jetzt gelöst, in dem der Key in die host_vars von linse kommt...